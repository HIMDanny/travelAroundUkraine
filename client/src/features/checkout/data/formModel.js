export default {
  firstName: {
    name: 'firstName',
    label: 'First Name',
    requiredErrorMsg: 'First name is required',
  },

  lastName: {
    name: 'lastName',
    label: 'Last Name',
    requiredErrorMsg: 'Last name is required',
  },
  email: {
    name: 'email',
    label: 'Email',
    requiredErrorMsg: 'Email is required',
    invalidErrorMsg: 'Please enter a valid email',
  },
  phone: {
    name: 'phone',
    label: 'Phone',
    requiredErrorMsg: 'Phone is required',
  },
  addressOne: {
    name: 'addressOne',
    label: 'Address line 1',
    requiredErrorMsg: 'Address is required',
  },
  addressTwo: {
    name: 'addressTwo',
    label: 'Address line 2',
  },
  city: {
    name: 'city',
    label: 'City',
    requiredErrorMsg: 'City is required',
  },
  state: {
    name: 'state',
    label: 'State/Province/Region',
  },
  zipcode: {
    name: 'zipcode',
    label: 'Zipcode',
    requiredErrorMsg: 'Zipcode is required',
  },
  country: {
    name: 'country',
    label: 'Country',
    requiredErrorMsg: 'Country is required',
  },
  cardType: {
    name: 'cardType',
    label: 'Card Type',
    requiredErrorMsg: 'Card Type is required',
  },
  nameOnCard: {
    name: 'nameOnCard',
    label: 'Name on Card',
    requiredErrorMsg: 'Name on card is required',
  },
  cardNumber: {
    name: 'cardNumber',
    label: 'Card Number',
    requiredErrorMsg: 'Card number is required',
    invalidErrorMsg: 'Please enter a valid card number ex: 1234123412341234',
  },
  expiryDate: {
    name: 'expiryDate',
    label: 'Expiry Date',
    requiredErrorMsg: 'Expiry date is required',
  },
  cvv: {
    name: 'cvv',
    label: 'CVV',
    requiredErrorMsg: 'CVV is required',
  },
};
