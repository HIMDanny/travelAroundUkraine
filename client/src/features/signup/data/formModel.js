export default {
  firstName: {
    name: 'firstName',
    label: 'First Name',
    requiredErrorMsg: 'First name is required',
  },

  lastName: {
    name: 'lastName',
    label: 'Last Name',
    requiredErrorMsg: 'Last name is required',
  },

  login: {
    name: 'login',
    label: 'Login',
    requiredErrorMsg: 'Login is required',
  },

  email: {
    name: 'email',
    label: 'Email',
    requiredErrorMsg: 'Email is required',
    invalidErrorMsg: 'Please enter a valid email',
  },

  password: {
    name: 'password',
    label: 'Password',
    requiredErrorMsg: 'Password is required',
  },

  retypePassword: {
    name: 'retypePassword',
    label: 'Re-type Password',
    requiredErrorMsg: 'Please retype your password',
    invalidErrorMsg: 'Passwords must match',
  },

  telephone: {
    name: 'telephone',
    label: 'Phone',
  },

  birthdate: {
    name: 'birthdate',
    label: 'Birthdate',
  },

  gender: {
    name: 'gender',
    label: 'Gender',
  },

  avatarUrl: {
    name: 'avatarUrl',
    label: 'Please upload your image',
  },
};
