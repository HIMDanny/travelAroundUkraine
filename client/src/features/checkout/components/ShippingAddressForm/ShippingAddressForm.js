/* eslint-disable react/prop-types */
import React from 'react';
import { Grid, Typography } from '@mui/material';
import { TextField } from 'formik-mui';
import { Field } from 'formik';
import {formModel} from '../../data';

const ShippingAddressForm = () => {
  const { addressOne, addressTwo, city, state, zipcode, country } = formModel;

  return (
    <>
      <Typography variant="h3" align="center" sx={{ mb: '20px' }}>
        Shipping Address
      </Typography>

      <Grid container spacing={3} mb={3}>
        <Grid item xs={12}>
          <Field component={TextField} required name={addressOne.name} label={addressOne.label} fullWidth />
        </Grid>

        <Grid item xs={12}>
          <Field component={TextField} name={addressTwo.name} label={addressTwo.label} fullWidth />
        </Grid>

        <Grid item xs={12} laptop={6}>
          <Field component={TextField} required name={city.name} label={city.label} fullWidth />
        </Grid>

        <Grid item xs={12} laptop={6}>
          <Field component={TextField} name={state.name} label={state.label} fullWidth />
        </Grid>

        <Grid item xs={12} laptop={6}>
          <Field component={TextField} required name={zipcode.name} label={zipcode.label} fullWidth />
        </Grid>

        <Grid item xs={12} laptop={6}>
          <Field component={TextField} required name={country.name} label={country.label} fullWidth />
        </Grid>
      </Grid>
    </>
  );
};

export default ShippingAddressForm;
