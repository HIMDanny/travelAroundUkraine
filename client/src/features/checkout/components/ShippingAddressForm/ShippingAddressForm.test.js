/* eslint-disable react/prop-types */
import { render } from '@testing-library/react';
import { Formik } from 'formik';
import ShippingAddressForm from './ShippingAddressForm';
import formModel from '../../data';

jest.mock('@mui/material/Typography', () => ({ children }) => <p>{children}</p>);
jest.mock('@mui/material/Grid', () => ({ children }) => <div>{children}</div>);

const FormikWrapper = ({ children }) => <Formik>{children}</Formik>;

describe('ShippingAddressForm snapshot test', () => {
  test('should ShippingAddressForm match snapshot', () => {
    const { asFragment } = render(
      <FormikWrapper>
        <ShippingAddressForm checkoutFormModel={formModel} />
      </FormikWrapper>
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
