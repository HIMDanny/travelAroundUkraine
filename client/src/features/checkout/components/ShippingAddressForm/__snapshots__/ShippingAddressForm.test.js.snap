// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ShippingAddressForm snapshot test should ShippingAddressForm match snapshot 1`] = `
<DocumentFragment>
  <p>
    Shipping Address
  </p>
  <div>
    <div>
      <div
        class="MuiFormControl-root MuiFormControl-fullWidth MuiTextField-root css-wb57ya-MuiFormControl-root-MuiTextField-root"
      >
        <label
          class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined MuiFormLabel-colorPrimary Mui-required MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined css-14s5rfu-MuiFormLabel-root-MuiInputLabel-root"
          data-shrink="false"
          for=":r0:"
          id=":r0:-label"
        >
          Address line 1
          <span
            aria-hidden="true"
            class="MuiFormLabel-asterisk MuiInputLabel-asterisk css-wgai2y-MuiFormLabel-asterisk"
          >
             *
          </span>
        </label>
        <div
          class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl css-md26zr-MuiInputBase-root-MuiOutlinedInput-root"
        >
          <input
            aria-invalid="false"
            class="MuiInputBase-input MuiOutlinedInput-input css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
            id=":r0:"
            name="addressOne"
            required=""
            type="text"
            value=""
          />
          <fieldset
            aria-hidden="true"
            class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
          >
            <legend
              class="css-yjsfm1"
            >
              <span>
                Address line 1 *
              </span>
            </legend>
          </fieldset>
        </div>
      </div>
    </div>
    <div>
      <div
        class="MuiFormControl-root MuiFormControl-fullWidth MuiTextField-root css-wb57ya-MuiFormControl-root-MuiTextField-root"
      >
        <label
          class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined MuiFormLabel-colorPrimary MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined css-14s5rfu-MuiFormLabel-root-MuiInputLabel-root"
          data-shrink="false"
          for=":r1:"
          id=":r1:-label"
        >
          Address line 2
        </label>
        <div
          class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl css-md26zr-MuiInputBase-root-MuiOutlinedInput-root"
        >
          <input
            aria-invalid="false"
            class="MuiInputBase-input MuiOutlinedInput-input css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
            id=":r1:"
            name="addressTwo"
            type="text"
            value=""
          />
          <fieldset
            aria-hidden="true"
            class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
          >
            <legend
              class="css-yjsfm1"
            >
              <span>
                Address line 2
              </span>
            </legend>
          </fieldset>
        </div>
      </div>
    </div>
    <div>
      <div
        class="MuiFormControl-root MuiFormControl-fullWidth MuiTextField-root css-wb57ya-MuiFormControl-root-MuiTextField-root"
      >
        <label
          class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined MuiFormLabel-colorPrimary Mui-required MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined css-14s5rfu-MuiFormLabel-root-MuiInputLabel-root"
          data-shrink="false"
          for=":r2:"
          id=":r2:-label"
        >
          City
          <span
            aria-hidden="true"
            class="MuiFormLabel-asterisk MuiInputLabel-asterisk css-wgai2y-MuiFormLabel-asterisk"
          >
             *
          </span>
        </label>
        <div
          class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl css-md26zr-MuiInputBase-root-MuiOutlinedInput-root"
        >
          <input
            aria-invalid="false"
            class="MuiInputBase-input MuiOutlinedInput-input css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
            id=":r2:"
            name="city"
            required=""
            type="text"
            value=""
          />
          <fieldset
            aria-hidden="true"
            class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
          >
            <legend
              class="css-yjsfm1"
            >
              <span>
                City *
              </span>
            </legend>
          </fieldset>
        </div>
      </div>
    </div>
    <div>
      <div
        class="MuiFormControl-root MuiFormControl-fullWidth MuiTextField-root css-wb57ya-MuiFormControl-root-MuiTextField-root"
      >
        <label
          class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined MuiFormLabel-colorPrimary MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined css-14s5rfu-MuiFormLabel-root-MuiInputLabel-root"
          data-shrink="false"
          for=":r3:"
          id=":r3:-label"
        >
          State/Province/Region
        </label>
        <div
          class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl css-md26zr-MuiInputBase-root-MuiOutlinedInput-root"
        >
          <input
            aria-invalid="false"
            class="MuiInputBase-input MuiOutlinedInput-input css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
            id=":r3:"
            name="state"
            type="text"
            value=""
          />
          <fieldset
            aria-hidden="true"
            class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
          >
            <legend
              class="css-yjsfm1"
            >
              <span>
                State/Province/Region
              </span>
            </legend>
          </fieldset>
        </div>
      </div>
    </div>
    <div>
      <div
        class="MuiFormControl-root MuiFormControl-fullWidth MuiTextField-root css-wb57ya-MuiFormControl-root-MuiTextField-root"
      >
        <label
          class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined MuiFormLabel-colorPrimary Mui-required MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined css-14s5rfu-MuiFormLabel-root-MuiInputLabel-root"
          data-shrink="false"
          for=":r4:"
          id=":r4:-label"
        >
          Zipcode
          <span
            aria-hidden="true"
            class="MuiFormLabel-asterisk MuiInputLabel-asterisk css-wgai2y-MuiFormLabel-asterisk"
          >
             *
          </span>
        </label>
        <div
          class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl css-md26zr-MuiInputBase-root-MuiOutlinedInput-root"
        >
          <input
            aria-invalid="false"
            class="MuiInputBase-input MuiOutlinedInput-input css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
            id=":r4:"
            name="zipcode"
            required=""
            type="text"
            value=""
          />
          <fieldset
            aria-hidden="true"
            class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
          >
            <legend
              class="css-yjsfm1"
            >
              <span>
                Zipcode *
              </span>
            </legend>
          </fieldset>
        </div>
      </div>
    </div>
    <div>
      <div
        class="MuiFormControl-root MuiFormControl-fullWidth MuiTextField-root css-wb57ya-MuiFormControl-root-MuiTextField-root"
      >
        <label
          class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined MuiFormLabel-colorPrimary Mui-required MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined css-14s5rfu-MuiFormLabel-root-MuiInputLabel-root"
          data-shrink="false"
          for=":r5:"
          id=":r5:-label"
        >
          Country
          <span
            aria-hidden="true"
            class="MuiFormLabel-asterisk MuiInputLabel-asterisk css-wgai2y-MuiFormLabel-asterisk"
          >
             *
          </span>
        </label>
        <div
          class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl css-md26zr-MuiInputBase-root-MuiOutlinedInput-root"
        >
          <input
            aria-invalid="false"
            class="MuiInputBase-input MuiOutlinedInput-input css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
            id=":r5:"
            name="country"
            required=""
            type="text"
            value=""
          />
          <fieldset
            aria-hidden="true"
            class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
          >
            <legend
              class="css-yjsfm1"
            >
              <span>
                Country *
              </span>
            </legend>
          </fieldset>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
